lighthouse-core/lib/i18n/i18n.js

```diff
@@ -358,143 +344,140 @@ function getRendererFormattedStrings(locale) {
 }
 
 /**
- * Register a file's UIStrings with i18n, return function to
- * generate the string ids.
- *
+ * Returns a function that generates `LH.IcuMessage` objects to localize the
+ * messages in `fileStrings` and the shared `i18n.UIStrings`.
  * @param {string} filename
  * @param {Record<string, string>} fileStrings
  */
 function createMessageInstanceIdFn(filename, fileStrings) {
-  /** @type {Record<string, string>} */
+  /**
+   * Combined so fn can access both caller's strings and i18n.UIStrings shared across LH.
+   * @type {Record<string, string>}
+   */
   const mergedStrings = {...UIStrings, ...fileStrings};
 
   /**
-   * Convert a message string & replacement values into an
-   * indexed id value in the form '{messageid} | # {index}'.
-   *
-   * @param {string} icuMessage
+   * Convert a message string and replacement values into an `LH.IcuMessage`.
+   * @param {string} message
    * @param {Record<string, string | number>} [values]
-   * */
-  const getMessageInstanceIdFn = (icuMessage, values) => {
-    const keyname = Object.keys(mergedStrings).find(key => mergedStrings[key] === icuMessage);
-    if (!keyname) throw new Error(`Could not locate: ${icuMessage}`);
+   * @return {LH.IcuMessage}
+   */
+  const getIcuMessageFn = (message, values) => {
+    const keyname = Object.keys(mergedStrings).find(key => mergedStrings[key] === message);
+    if (!keyname) throw new Error(`Could not locate: ${message}`);
 
     const filenameToLookup = keyname in fileStrings ? filename : __filename;
     const unixStyleFilename = path.relative(LH_ROOT, filenameToLookup).replace(/\\/g, '/');
-    const icuMessageId = `${unixStyleFilename} | ${keyname}`;
-    const icuMessageInstances = _icuMessageInstanceMap.get(icuMessageId) || [];
+    const i18nId = `${unixStyleFilename} | ${keyname}`;
 
-    let indexOfInstance = icuMessageInstances.findIndex(inst => isDeepEqual(inst.values, values));
-    if (indexOfInstance === -1) {
-      icuMessageInstances.push({icuMessageId, icuMessage, values});
-      indexOfInstance = icuMessageInstances.length - 1;
-    }
-
-    _icuMessageInstanceMap.set(icuMessageId, icuMessageInstances);
-
-    return `${icuMessageId} # ${indexOfInstance}`;
```

Comment:
we can eliminate the `# ${index}` part of this because we keep the values to format in the object itself now right?